src/qtcore/qml/QObject.js

```diff
@@ -26,5 +29,7 @@ function QObject(parent) {
 
         if (this.$parent && this.$parent.$tidyupList)
             this.$parent.$tidyupList.splice(this.$parent.$tidyupList.indexOf(this), 1);
+
+        this.parent = undefined; // must do this. => 1) parent will be notified and erase object from it's children. 2) DOM node will be removed.
```

Comment:
PTAL at change at https://github.com/qmlweb/qmlweb/commit/ebf0b623d298d2db8c95a0119ce4a980b5d882e1#diff-3a670166eb3912cd466ef8ce0543589d

Are those related? Should we do both or only one of them?